engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -873,17 +873,31 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
   override def queryWorkflows(queryParameters: WorkflowQueryParameters)
                              (implicit ec: ExecutionContext): Future[WorkflowQueryResponse] = {
-    val action = dataAccess.queryWorkflowExecutions(queryParameters).result
-    runTransaction(action) map { workflows =>
-      WorkflowQueryResponse(workflows map { workflow =>
-        WorkflowQueryResult(
-          id = workflow.workflowExecutionUuid,
-          name = workflow.name,
-          status = workflow.status,
-          start = new DateTime(workflow.startDt),
-          end = workflow.endDt map { new DateTime(_) })
-      })
-    }
+    val workflowExecutionsQuery = dataAccess.queryWorkflowExecutions(queryParameters).result
+
+    val workflowExecutionCountQuery = dataAccess.countWorkflowExecutions(queryParameters).result
+
+    for {
+      count <- runTransaction(workflowExecutionCountQuery)
+      workflowExecutions <- runTransaction(workflowExecutionsQuery)
+    } yield WorkflowQueryResponse(
```

Comment:
The two separate transactions might yield inconsistent results.  Perhaps compose in the DBIO monad and wrap the composite action with a transaction?

```scala

    case class CountAndWorkflowExecutions(count: Int, workflowExecutions: Seq[WorkflowExecution])

    val action = for {
      workflowExecutions <- dataAccess.queryWorkflowExecutions(queryParameters).result
      count <- dataAccess.countWorkflowExecutions(queryParameters).result
    } yield CountAndWorkflowExecutions(count, workflowExecutions)
    
    runTransaction(action) map { case c => 
...
```